compiler/src/codegen/transl_anf.ml

```diff
@@ -88,6 +88,150 @@ let worklist_pop () =
+      | MError(e, is) -> MError(e, List.map apply_allocation_to_imm is)
+      | MCallIndirect(imm, is) -> MCallIndirect(apply_allocation_to_imm imm, List.map apply_allocation_to_imm is)
+      | MIf(c, t, f) -> MIf(apply_allocation_to_imm c, apply_allocation_to_block t, apply_allocation_to_block f)
+      | MWhile(b1, b2) -> MWhile(apply_allocation_to_block b1, apply_allocation_to_block b2)
+      | MSwitch(v, bs, d) -> MSwitch(apply_allocation_to_imm v, List.map (fun (t, b) -> (t, apply_allocation_to_block b)) bs, apply_allocation_to_block d)
+      | MPrim2(pop, i1, i2) -> MPrim2(pop, apply_allocation_to_imm i1, apply_allocation_to_imm i2)
+      | MStore(bs) -> MStore(List.map (fun (b, bk) -> (apply_allocation_to_bind b, apply_allocations allocs bk)) bs)
+      | MAllocate(x) -> MAllocate(x)
+      | MDrop -> MDrop
+      | MTracepoint(x) -> MTracepoint(x)
+end
+
+let run_register_allocation (instrs: Mashtree.instr list) =
+  let uniq l =
+    let rec tail_uniq a l =
+      match l with
+        | [] -> a
+        | hd::tl -> tail_uniq (hd::a) (List.filter (fun x -> x != hd) tl) in
+    tail_uniq [] l in
+  let bind_live_local = function
+    | MLocalBind(n) -> [Int32.to_int n]
+    | _ -> []
+  in
+  let imm_live_local = function
+    | MImmBinding(MLocalBind(n)) -> [Int32.to_int n]
+    | _ -> []
+  in
+  let rec live_locals = function
+    | MImmediate(imm)
+    | MTagOp(_, _, imm)
+    | MArityOp(_, _, imm)
+    | MPrim1(_, imm)
+    | MTupleOp(_, imm)
+    | MBoxOp(_, imm)
+    | MArrayOp(_, imm)
+    | MRecordOp(_, imm)
+    | MAdtOp(_, imm) -> imm_live_local imm
+    | MCallKnown(_, is)
+    | MError(_, is) -> List.concat (List.map imm_live_local is)
+    | MCallIndirect(imm, is) -> (List.concat (List.map imm_live_local is)) @ (imm_live_local imm)
+    | MIf(c, t, f) -> (imm_live_local c) @ (block_live_locals t) @ (block_live_locals f)
+    | MWhile(b1, b2) -> (block_live_locals b1) @ (block_live_locals b2)
+    | MSwitch(v, bs, d) -> (imm_live_local v) @ (List.concat (List.map (fun (_, b) -> block_live_locals b) bs)) @ (block_live_locals d)
+    | MPrim2(_, i1, i2) -> (imm_live_local i1) @ (imm_live_local i2)
+    | MStore(bs) -> List.concat (List.map (fun (b, bk) -> (bind_live_local b) @ (live_locals bk)) bs)
+    | MAllocate _
+    | MTracepoint _
+    | MDrop -> []
+  and block_live_locals b = List.concat (List.map live_locals b)
+  in
+  (** Mapping from (instruction index) -> (used locals) *)
+  let instr_live_sets = List.mapi (fun i instr -> (i, uniq @@ live_locals instr)) instrs in
+  let num_locals =
+    let rec help acc = function
+      | [] -> acc
+      | hd::tl when hd > acc -> help hd tl
+      | _::tl -> help acc tl in
+    1 + (help 0 (List.map (fun (_, lst) -> help 0 lst) instr_live_sets)) in
+  (* Printf.eprintf "Live sets:\n";
+  List.iter (fun (i, items) -> Printf.eprintf "%d -> [%s]\n" i (BatString.join ", " (List.map string_of_int items))) instr_live_sets; *)
```

Comment:
```suggestion
```

Reply:
Not sure why you want to remove these? They're useful for debugging the register allocation